inference-engine/src/transformations/src/transformations/convert_opset1_to_legacy/convert_proposal_to_proposal_ie.cpp

```diff
@@ -8,58 +8,78 @@
 #include <vector>
 
 #include <ngraph/opsets/opset1.hpp>
-
+#include <ngraph/opsets/opset4.hpp>
+#include <ngraph/pattern/op/wrap_type.hpp>
 #include <ngraph_ops/proposal_ie.hpp>
 #include <ngraph/rt_info.hpp>
 
-ngraph::pass::ConvertProposalToLegacyMatcher::ConvertProposalToLegacyMatcher() {
-    auto input_0 = std::make_shared<pattern::op::Label>(element::f32, Shape{1, 1, 1, 1});
-    auto input_1 = std::make_shared<pattern::op::Label>(element::f32, Shape{1, 1, 1, 1});
-    auto input_2 = std::make_shared<pattern::op::Label>(element::f32, Shape{3});
+bool convert_to_proposal_ie(std::shared_ptr<ngraph::op::v0::Proposal> proposal, bool add_second_out = false) {
+    ngraph::Output<ngraph::Node> last;
 
-    ngraph::op::ProposalAttrs attr = {};
+    ngraph::NodeVector ops_to_replace, new_ops;
+    ops_to_replace.push_back(proposal);
 
-    auto proposal = std::make_shared<ngraph::opset1::Proposal>(input_0, input_1, input_2, attr);
+    if (auto reshape = std::dynamic_pointer_cast<ngraph::opset1::Reshape>(
+            proposal->input_value(2).get_node_shared_ptr())) {
+        auto input_shape = reshape->get_input_shape(0);
```

Comment:
You use this shape only for getting its rank. Also you don't have check that shape is static.
I suggest to get partial shape instead and then check rank.get_length()

Reply:
Rewritten if statement, now I get partial shape and check not rank but if the shape is acceptable. Please review.